utils/src/main/scala/za/co/absa/enceladus/utils/performance/PerformanceMetricTools.scala

```diff
@@ -28,15 +28,62 @@ object PerformanceMetricTools {
 
   private val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
+  /**
+    * Adds general job information to Atum's metadata that will end up in an info file.
+    *
+    * The method should be ran at the beginning of a job so that this general information is available for debugging.
+    *
+    * @param spark         A Spark Session
+    * @param optionPrefix  A prefix for all performance metrics options, e.g. 'std' for Standardization and 'conform' for Conformance
+    * @param inputPath     A path to an input directory of the job
+    * @param outputPath    A path to an output directory of the job
+    * @param loginUserName A login user name who performed the job
+    *
+    **/
+  def addJobInfoToAtumMetadata(optionPrefix: String,
+                                inputPath: String,
+                                outputPath: String,
+                                loginUserName: String,
+                                cmdLineArgs: String)
+                               (implicit spark: SparkSession): Unit = {
+    // Spark job configuration
+    val sc = spark.sparkContext
+
+    // The number of executors minus the driver
+    val numberOfExecutors = sc.getExecutorMemoryStatus.keys.size - 1
+
+    val fsUtils = new FileSystemVersionUtils(spark.sparkContext.hadoopConfiguration)
+    // Directory sizes and size ratio
+    val inputDirSize = fsUtils.getDirectorySize(inputPath)
+
+    addSparkConfig(optionPrefix, "spark.driver.memory", "driver_memory")
+    addSparkConfig(optionPrefix, "spark.driver.cores", "driver_cores")
+    addSparkConfig(optionPrefix, "spark.driver.memoryOverhead", "driver_memory_overhead")
+    addSparkConfig(optionPrefix, "spark.executor.memory", "executor_memory")
+    addSparkConfig(optionPrefix, "spark.executor.cores", "executor_cores")
+    addSparkConfig(optionPrefix, "spark.executor.memoryOverhead", "executor_memory_overhead")
+    addSparkConfig(optionPrefix, "spark.submit.deployMode", "yarn_deploy_mode")
+    addSparkConfig(optionPrefix, "spark.master", "spark_master")
+
+    Atum.setAdditionalInfo(s"${optionPrefix}_cmd_line_args" -> cmdLineArgs)
+    Atum.setAdditionalInfo(s"${optionPrefix}_input_dir" -> inputPath)
+    Atum.setAdditionalInfo(s"${optionPrefix}_output_dir" -> outputPath)
+    Atum.setAdditionalInfo(s"${optionPrefix}_input_dir_size" -> inputDirSize.toString)
+    Atum.setAdditionalInfo(s"${optionPrefix}_enceladus_version" -> ProjectMetadataTools.getEnceladusVersion)
+    Atum.setAdditionalInfo(s"${optionPrefix}_application_id" -> spark.sparkContext.applicationId)
+    Atum.setAdditionalInfo(s"${optionPrefix}_username" -> loginUserName)
+    Atum.setAdditionalInfo(s"${optionPrefix}_executors_num" -> s"$numberOfExecutors")
```

Comment:
Isn't a number of executors also a `spakrConfig` option?

Reply:
It's not always there. The way it gets the number of executors now is a "safer way".

Most of the listed options are absent at runtime unless explicitly specified in `spark-submit`. For instance, if the number of cores is not specified, this configuration key will be missing: `spark.executor.cores`. I assume that Spark will use the default value which is `1` for this setting.

Reply:
Cool